store/src/directory.rs

```diff
@@ -68,6 +119,101 @@ where
     }
 }
 
+/// Filesystems supported by the linux kernel that also support extended attributes
+enum FileSystemXattrSupport {
+    Btrfs,
+    Ext2,
+    Ext3,
+    Ext4,
+    F2fs,
+    Lustre,
+    Jfs,
+    Ocfs2,
+    Orangefs,
+    Reiser4,
+    Reiserfs,
+    Squashfs,
+    Ubifs,
+    Xfs,
+    Yaffs2,
+    Zfs,
+}
+
+impl FromStr for FileSystemXattrSupport {
+    type Err = StoreError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "btrfs" => Ok(FileSystemXattrSupport::Btrfs),
+            "ext2" => Ok(FileSystemXattrSupport::Ext2),
+            "ext3" => Ok(FileSystemXattrSupport::Ext3),
+            "ext4" => Ok(FileSystemXattrSupport::Ext4),
+            "f2fs" => Ok(FileSystemXattrSupport::F2fs), // check
+            "lustre" => Ok(FileSystemXattrSupport::Lustre), // check
+            "jfs" => Ok(FileSystemXattrSupport::Jfs),
+            "ocfs2" => Ok(FileSystemXattrSupport::Ocfs2), // check
+            "orangefs" => Ok(FileSystemXattrSupport::Orangefs), // check
+            "reiser4" => Ok(FileSystemXattrSupport::Reiser4),
+            "reiser" => Ok(FileSystemXattrSupport::Reiserfs),
+            "squashfs" => Ok(FileSystemXattrSupport::Squashfs),
+            "ubifs" => Ok(FileSystemXattrSupport::Ubifs),
+            "xfs" => Ok(FileSystemXattrSupport::Xfs),
+            "yaffs2" => Ok(FileSystemXattrSupport::Yaffs2), // check
+            "zfs" => Ok(FileSystemXattrSupport::Zfs),
+            _ => Err(StoreError::Configuration(format!("Unsupported filesystem"))),
+        }
+    }
+}
```

Comment:
so, usually, I think a better approach is "testing for features" - so instead of hardcoding this info here, you can just try and set xattrs, if the doesn't work we can safely assume it's not supported